Add comprehensive image normalization documentation for wandb.Image DOCS-1016

mdlinville opened the pull request:
## Summary

This PR addresses [DOCS-1016](https://wandb.atlassian.net/browse/DOCS-1016) and the related GitHub issue [wandb/wandb#7377](https://github.com/wandb/wandb/issues/7377) by adding comprehensive documentation for `wandb.Image` normalization behavior.

This is part of a three-part project. The other two PRs are:

- New example Colab notebook: https://github.com/wandb/examples/pull/609
- `wandb/wandb` change: https://github.com/wandb/wandb/pull/10380

Ready for review.

## Changes Made

### Documentation Updates (`docs` repository)

1. **Enhanced `Image.md` reference documentation** (`content/en/ref/python/sdk/data-types/Image.md`):
   - Added detailed "Image Normalization" section with sub-sections:
     - When Normalization is Applied (PyTorch tensors, NumPy arrays vs PIL Images, file paths)
     - Normalization Algorithm (explaining `_guess_and_rescale_to_0_255` logic)
     - Examples of Normalization Effects (with code snippets for different input ranges)
     - Best Practices
     - Common Pitfalls

2. Added Image normalization guide (new page) with some more details.
3. **Updated media logging guide** (`content/en/guides/models/track/log/media.md`):
   - Modified existing text to reference the new detailed normalization section

4. **Translated documentation** (Japanese and Korean):
   - Updated `content/ja/ref/python/data-types/image.md` and `content/ko/ref/python/data-types/image.md` with translated normalization sections
   - Updated media guides in both languages to reference the new sections
   - Fixed broken `relref` links that were pointing to incorrect paths

### SDK Documentation Updates (`wandb/wandb` repository)

See https://github.com/wandb/wandb/pull/10380

## Problem Solved

The original issue was that users found the existing documentation about image normalization to be "cryptic" and unclear. Users were wasting computation time due to misunderstandings about:
- When normalization occurs
- How the normalization algorithm works
- What input ranges are supported
- How to avoid unwanted normalization

## Testing

- Hugo build completes successfully with no errors
- All `relref` links are properly resolved
- Documentation is available in English, Japanese, and Korean
- Developed the notebook in  https://github.com/wandb/examples/pull/609 and ran the cells

## Related PRs

- Companion PR in `wandb/wandb` repository: [wandb/wandb#10380](https://github.com/wandb/wandb/pull/10380)
- Companior PR in `wandb/examples` repository: https://github.com/wandb/examples/pull/609

## JIRA Reference

- [DOCS-1016](https://wandb.atlassian.net/browse/DOCS-1016): "Productboard: [Docs] Better explanation for the section: when logging a PyTorch tensor as a PIL Image"

<!-- preview-links-comment -->
📄 **[View preview links for changed pages](https://github.com/wandb/docs/pull/1559#issuecomment-3215338217)**

[DOCS-1016]: https://wandb.atlassian.net/browse/DOCS-1016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOCS-1016]: https://wandb.atlassian.net/browse/DOCS-1016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mdlinville committed:
test: temporarily use mock preview URL for testing path matching

mdlinville committed:
revert: remove mock URL test changes

mdlinville committed:
Add comprehensive image normalization documentation

- Add detailed 'Image Normalization' section to Image.md reference explaining the normalization algorithm
- Document when normalization is applied (PyTorch tensors, NumPy arrays) vs when it's not (PIL Images, file paths)
- Provide specific examples showing [0,1], [-1,1], and other range normalization effects
- Add examples demonstrating how to avoid normalization using PIL Images or normalize=False
- Include best practices and common pitfalls
- Update media logging guide to reference the detailed normalization documentation
- Add comprehensive translations in Japanese and Korean

Addresses DOCS-1016 - resolves user confusion about PyTorch tensor normalization behavior

mdlinville commented:
**Update**: Companion PR in `wandb/wandb` repository: [wandb/wandb#10380](https://github.com/wandb/wandb/pull/10380)

mdlinville committed:
Fix broken relref links in Japanese and Korean media guides

mdlinville commented on content/en/guides/models/track/log/media.md:
````diff
@@ -58,7 +58,7 @@ with wandb.init(project="image-log-example") as run:
 ```
 
-We assume the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, we convert them to integers between `0` and `255`. If you want to normalize your images differently, you can specify the [`mode`](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes) manually or just supply a [`PIL.Image`](https://pillow.readthedocs.io/en ...
+We assume the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, we automatically normalize them to integers between `0` and `255` using our normalization algorithm. For detailed information about how normalization works with PyTorch tensors and NumPy arrays, see the [Image Normalization section]({{< relref "/ref/python/sdk/data-types/ ...
````
Avoid "we". Reword this to be more imperative.

mdlinville commented on content/en/ref/python/sdk/data-types/Image.md:
````diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
     run.log({"examples": examples})
 ``` 
 
+## Image Normalization
````
Use sentence case for headings

mdlinville commented on content/en/ref/python/sdk/data-types/Image.md:
```diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
 
+## Image Normalization
+
+When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This normalization is designed to handle common image data formats and ensure proper display.
```
```suggestion
When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This normalization is designed to handle and ensure proper display of common image formats.
```

mdlinville commented on content/en/ref/python/sdk/data-types/Image.md:
````diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
+image = wandb.Image(tensor_0_1, normalize=False, caption="Normalization disabled")
+```
+
+### Best Practices
````
Recommendations

mdlinville commented on content/en/ref/python/sdk/data-types/Image.md:
```diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
+3. **For debugging**: Use `normalize=False` to see the raw values (they will be clipped to [0, 255])
+4. **For precise control**: Use PIL Images when you need exact pixel values
+
+### Common Pitfalls
```
Troubleshooting

mdlinville committed:
docs: address review feedback - use sentence case, avoid 'we', improve headings

mdlinville commented on content/en/guides/models/track/log/media.md:
````diff
@@ -58,7 +58,7 @@ with wandb.init(project="image-log-example") as run:
 ```
 
-We assume the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, we convert them to integers between `0` and `255`. If you want to normalize your images differently, you can specify the [`mode`](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes) manually or just supply a [`PIL.Image`](https://pillow.readthedocs.io/en ...
+The system assumes the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, the system automatically normalizes them to integers between `0` and `255` using the normalization algorithm. For detailed information about how normalization works with PyTorch tensors and NumPy arrays, see the [Image Normalization section]({{< relref "/ref/pyth ...
````
```suggestion
The system assumes the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, the system automatically normalizes them to integers between `0` and `255`. For detailed information about normalization with PyTorch tensors and NumPy arrays, see the [Image normalization section]({{< relref "/ref/python/sdk/data-types/image.md#image-normalization" >}}) in the Image SDK reference. To normalize your images differently, you can specify the [`mode`](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes) manually or supply a [`PIL.Image`](https://pillow.readthedocs.io/en/stable/reference/Image.html), as described in the "Logging PIL Images" tab.
```

mdlinville commented on content/en/ref/python/sdk/data-types/Image.md:
```diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
+
+When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This normalization is designed to handle and ensure proper display of common image formats.
+
+### When Normalization is Applied
```
Use sentence case for all headings

mdlinville committed:
docs: address additional review feedback - use sentence case for headings, improve text clarity

ngrayluna reviewed (commented):
Thanks! Couple of questions, comments:
* Did you test out the code snippets generated?
* You'll want to make a PR for changes added in the ref page

ngrayluna commented on content/en/ref/python/sdk/data-types/Image.md:
````diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
     run.log({"examples": examples})
 ``` 
 
+## Image normalization
````
Remove "#" signs for docs intended for docstrings. They don't render correctly.

ngrayluna commented on content/en/ref/python/sdk/data-types/Image.md:
````diff
@@ -103,6 +103,103 @@ with wandb.init() as run:
+image = wandb.Image(tensor_0_1, normalize=False, caption="Normalization disabled")
+```
+
+### Recommendations
````
These sections look like they'd be better suited for the dev guide, not the reference guide.

ngrayluna reviewed (commented):
This content is better suited for the dev guide.  You want to keep the docstrings as concise as possible. They should include code examples, but that's it; (i.e. no markdown sections , etc.)

ngrayluna committed:
Artifacts: When to use save() vs .log() (#1557)

Description
-----------
Adds quick blurb on "best practice"/recommendation by Eng on when to use
which.


Related issues
-----------
- Fixes
https://wandb.atlassian.net/jira/software/projects/DOCS/boards/28?selectedIssue=DOCS-1087

knisar committed:
Updated new view for lineage and dataset mismatch in keras tutorials (#1541)

mdlinville committed:
docs(weave): fix broken relref links in Japanese and Korean media guides

mdlinville commented:
Test comment to verify GitHub MCP server is working properly.

mdlinville committed:
docs: address review feedback - move detailed normalization content to dev guide, simplify reference

mdlinville committed:
feat: add image normalization demo notebook for testing and user examples

mdlinville committed:
refactor: move detailed normalization docs to dedicated guide and simplify reference docs

mdlinville committed:
docs: add visual contrast effects and guidance on when to use different normalization approaches

mdlinville committed:
docs: add improved image normalization demo notebook with detailed explanations

mdlinville committed:
docs: update notebook with detailed explanations for all examples

mdlinville committed:
docs: use --quiet --upgrade flags for pip install to reduce output noise

mdlinville committed:
docs: add reassuring note about import cell success

mdlinville committed:
docs: add explanation about expected deprecation warning in Example 2

mdlinville committed:
docs: add reassuring note about import cell success

mdlinville committed:
docs: restore --quiet --upgrade flags to pip install command

mdlinville committed:
docs: remove --upgrade flag from pip install to avoid dependency conflicts

mdlinville committed:
docs: remove notebook (moved to examples repo PR #609)

mdlinville committed:
fix: correct GITHUB_TOKEN parameter name in calibreapp image actions workflow

mdlinville committed:
Merge branch 'main' into DOCS-1016

mdlinville commented on content/en/guides/models/track/log/image-normalization.md:
```diff
@@ -0,0 +1,139 @@
+description: "Learn how wandb.Image handles normalization for different input types and how to control this behavior"
+---
+
+# Image Normalization Guide
```
```suggestion
```

mdlinville commented on content/en/guides/models/track/log/image-normalization.md:
```diff
@@ -0,0 +1,139 @@
+
+# Image Normalization Guide
+
+When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This guide explains how normalization works and how to control it.
```
```suggestion
When you pass PyTorch tensors or NumPy arrays to `wandb.Image`, the pixel values are automatically normalized to the range [0, 255] unless you set `normalize=False`. This guide explains how image normalization works and how to control it.
```

mdlinville commented on content/en/guides/models/track/log/image-normalization.md:
```diff
@@ -0,0 +1,139 @@
+
+Normalization is **NOT** applied to:
+- **PIL Images** (passed as-is)
+- **File paths** (loaded as-is)
```
Consider converting this to a table using ✅ and ❌ .

mdlinville commented on content/en/guides/models/track/log/image-normalization.md:
````diff
@@ -0,0 +1,139 @@
+3. **For any other range**: Values are clipped to [0, 255] and converted to uint8
+   ```python
+   normalized_data = data.clip(0, 255).astype(np.uint8)
+   ```
````
Change these to bullets

mdlinville commented on content/en/guides/models/track/log/image-normalization.md:
```diff
@@ -0,0 +1,139 @@
+
+This "stretches" the visual range, making differences between pixel values more pronounced. This is particularly useful for highlighting subtle patterns in machine learning data, but if you want less contrast, consider preprocessing your data to a [0, 1] range before logging.
+
+### Example 3: Avoiding normalization with PIL Images
```
```suggestion
### Example 3: Avoid normalization with PIL Images
```

mdlinville commented on content/en/guides/models/track/log/image-normalization.md:
```diff
@@ -0,0 +1,139 @@
+This "stretches" the visual range, making differences between pixel values more pronounced. This is particularly useful for highlighting subtle patterns in machine learning data, but if you want less contrast, consider preprocessing your data to a [0, 1] range before logging.
+
+### Example 3: Avoiding normalization with PIL Images
+
```
```suggestion
Normalization is not applied to PIL Images.

```

mdlinville commented on content/en/guides/models/track/log/image-normalization.md:
````diff
@@ -0,0 +1,139 @@
+```
+
+### Example 4: Using normalize=False
+
````
```suggestion
To explicitly turn off image normalization without converting the image, set `normalize=False`.

```

mdlinville commented on content/en/guides/models/track/log/image-normalization.md:
```diff
@@ -0,0 +1,139 @@
+- Your data is in standard ranges ([0, 1] or [-1, 1])
+- You want the system to handle the conversion automatically
+
+## Best practices
```
Combine the Best practices sectino with the Common issues section and call it Troubleshooting. You will need to adjust the contents a bit.

mdlinville commented on content/en/guides/models/track/log/media.md:
````diff
@@ -58,7 +58,7 @@ with wandb.init(project="image-log-example") as run:
 ```
 
-We assume the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, we convert them to integers between `0` and `255`. If you want to normalize your images differently, you can specify the [`mode`](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes) manually or just supply a [`PIL.Image`](https://pillow.readthedocs.io/en ...
+The system assumes the image is gray scale if the last dimension is 1, RGB if it's 3, and RGBA if it's 4. If the array contains floats, the system automatically normalizes them to integers between `0` and `255`. For detailed information about normalization with PyTorch tensors and NumPy arrays, see the [Image Normalization Guide]({{< relref "/guides/models/track/log/image-normalization.md" >}}).  ...
````
Use "W&B" instead of "the system". Use sentence case with initial cap for the link text.

mdlinville committed:
docs: address reviewer feedback - combine sections, fix terminology, improve formatting

mdlinville committed:
docs: remove # characters from headings in reference documentation

mdlinville committed:
Apply suggestions from code review

mdlinville committed:
docs: remove remaining # characters from reference documentation headings

mdlinville committed:
docs: remove # characters from English reference documentation headings

mdlinville committed:
docs: improve normalization table with matrix format for better clarity

mdlinville committed:
docs: convert numbered normalization algorithm steps to bullet points for better clarity

mdlinville committed:
docs: remove accidentally committed test notebook

ngrayluna commented on content/en/guides/models/track/log/image-normalization.md:
```diff
@@ -0,0 +1,141 @@
+
+| Input Type | Format | Normalization Applied | Notes |
+|------------|--------|----------------------|-------|
+| **PyTorch tensors** | `(channel, height, width)` | ✅ Yes | Automatically normalized to [0, 255] range |
```
Remove emojis

ngrayluna commented on content/ja/guides/integrations/pytorch.md:
```diff
@@ -49,7 +49,7 @@ for batch_idx, (data, target) in enumerate(train_loader):
 
 ## 画像とメディアのログ
 
-画像データを持つ PyTorch `Tensors` を [`wandb.Image`]({{< relref path="/ref/python/data-types/image.md" lang="ja" >}}) に渡すことができ、[`torchvision`](https://pytorch.org/vision/stable/index.html) のユーティリティが自動的に画像に変換します。
```
Suggest undoing changes to JA and KO docs. They will get overwritten. Also, keeps the PR cleaner for review.

ngrayluna commented on content/en/ref/python/sdk/data-types/Image.md:
```diff
@@ -9,10 +9,10 @@ data_type_classification: class
 
 
-## <kbd>class</kbd> `Image`
+<kbd>class</kbd> `Image`
```
Formatting is correct. Undo changes. (Note: Anything in content/en/ref/python/* autogenerated. To make changes, make a PR in wandb/wandb).

mdlinville commented:
This PR is too low quality to keep, I'll make another attempt once we are on Mintlify.
